meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClientCache.java

```diff
@@ -70,10 +69,16 @@ public ZkMetaClientCache(ZkMetaClientConfig config, MetaClientCacheConfig cacheC
         }
     }
 
+    /**
+     * Get data for a given key.
+     * If datacache is enabled, will fetch for cache. If it doesn't exist
+     * returns null (for when initial populating cache is in progress).
+     * @param key key to identify the entry
+     * @return data for the key
+     */
     @Override
     public T get(final String key) {
         if (_cacheData) {
-            getDataCacheMap().computeIfAbsent(key, k -> _cacheClient.readData(k, true));
             return getDataCacheMap().get(key);
```

Comment:
shall we differentiate if key is not in ZK, or it is in ZK but not populated to cache yet?

Reply:
I am not too sure how to do this without pinging zookeeper. If it's not populated yet how do we know if it doesn't actually exist in zookeeper or it's being populated without checking zk?